presto-spi/src/main/java/io/prestosql/spi/type/TypeOperators.java

```diff
@@ -116,20 +117,29 @@ public MethodHandle getIndeterminateOperator(Type type, InvocationConvention cal
         return getOperatorAdaptor(type, callingConvention, OperatorType.INDETERMINATE).get();
     }
 
-    public MethodHandle getComparisonOperator(Type type, InvocationConvention callingConvention)
+    public MethodHandle getComparisonUnorderedLastOperator(Type type, InvocationConvention callingConvention)
     {
         if (!type.isOrderable()) {
             throw new UnsupportedOperationException(type + " is not orderable");
         }
-        return getOperatorAdaptor(type, callingConvention, COMPARISON).get();
+        return getOperatorAdaptor(type, callingConvention, COMPARISON_UNORDERED_LAST).get();
+    }
+
+    public MethodHandle getComparisonUnorderedFirstOperator(Type type, InvocationConvention callingConvention)
+    {
+        if (!type.isOrderable()) {
+            throw new UnsupportedOperationException(type + " is not orderable");
+        }
+        return getOperatorAdaptor(type, callingConvention, COMPARISON_UNORDERED_FIRST).get();
     }
 
     public MethodHandle getOrderingOperator(Type type, SortOrder sortOrder, InvocationConvention callingConvention)
     {
         if (!type.isOrderable()) {
             throw new UnsupportedOperationException(type + " is not orderable");
         }
-        return getOperatorAdaptor(type, Optional.of(sortOrder), callingConvention, COMPARISON).get();
+        OperatorType comparisonType = sortOrder.isNullsFirst() ? COMPARISON_UNORDERED_LAST : COMPARISON_UNORDERED_FIRST;
+        return getOperatorAdaptor(type, Optional.of(sortOrder), callingConvention, comparisonType).get();
```

Comment:
So it's interesting that the nulls ordering is handled through an operator adapter, rather than a separate operator. Would it make sense to have the unordered first/last also be an InvocationConvention as well rather than a separate operator?

Reply:
The invocation convention is really about the shape and capabilitties of the callsite, and not really about the desired target linkage method.  Also, I expect that would be a much bigger change.